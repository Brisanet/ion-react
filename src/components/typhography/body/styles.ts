import stitches from '../../../stitches.config';
const { styled } = stitches;

export const Body = styled('p', {
  margin: 0,
  padding: 0,
  variants: {
    color: {
      primary: { color: '$neutral7' },
      secondary: { color: '$neutral6' },
      'dark-primary': { color: '$neutral3' },
      'dark-secondary': { color: '$neutral4' },
    },
    weight: {
      normal: { fontWeight: '400' },
      medium: { fontWeight: '600' },
      bold: { fontWeight: '700' },
    },
    style: { normal: { fontStyle: 'normal' }, italic: { fontStyle: 'italic' } },
    size: {
      small: { fontSize: '14px', lineHeight: '20px' },
      normal: { fontSize: '16px', lineHeight: '24px' },
    },
  },
});
