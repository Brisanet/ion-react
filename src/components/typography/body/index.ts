export * from './body';
export * from './styles';
