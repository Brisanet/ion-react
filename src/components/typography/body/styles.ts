import { css, styled } from 'styled-components';
import {
  IonBodyInterface,
  IonTypographyColorScheme,
  IonTypographyStyle,
  IonTypographyWeight,
} from '../typography.types';
import { defaultColor } from '../utils-typography';

type BodyStyleProps = {
  $color: IonTypographyColorScheme;
  $size: IonBodyInterface['size'];
  $style: IonTypographyStyle;
  $weight: IonTypographyWeight;
};

export const weights = {
  normal: 400,
  medium: 600,
  bold: 700,
};

export const sizes = {
  small: 14,
  normal: 16,
};

export const fontStyle = {
  normal: 'normal',
  italic: 'italic',
};

export const Body = styled.p<BodyStyleProps>`
  ${({ $color, $size, $style, $weight }) => css`
    margin: 0;
    padding: 0;
    color: ${defaultColor('body')[$color].color};
    ${$size && sizes[$size]};
    font-weight: ${weights[$weight]};
    font-style: ${fontStyle[$style]};
  `};
`;
