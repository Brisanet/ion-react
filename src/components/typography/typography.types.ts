export type IonTypographyStyle = 'normal' | 'italic';

export type IonTypographyColorScheme =
  | 'primary'
  | 'secondary'
  | 'dark-primary'
  | 'dark-secondary';

interface BaseIonTypography {
  label: string;
  colorScheme?: IonTypographyColorScheme;
}

export type IonTypographySize = 'small' | 'medium' | 'normal';
export type IonTypographyWeight = 'normal' | 'medium' | 'bold';

type IonBodySize<TType> = TType extends 'medium' ? never : TType;
export type IonHeadingWeight<TType> = TType extends 'normal' ? never : TType;

export interface IonBodyInterface extends BaseIonTypography {
  size?: IonBodySize<IonTypographySize>;
  weight?: IonTypographyWeight;
  style?: IonTypographyStyle;
}

export interface IonHeadingInterface extends BaseIonTypography {
  type: 'h1' | 'h2' | 'h3' | 'h4';
  size?: IonTypographySize;
  weight?: IonHeadingWeight<IonTypographyWeight>;
}
