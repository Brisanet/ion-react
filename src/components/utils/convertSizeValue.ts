/**
 * Get a size value based on its type.
 * @param {string | number} value - The size value.
 * @returns The value itself if it's a string, or the value in pixels if it is a number.
 **/
export const convertSizeValue = (value: string | number): string =>
  typeof value === 'number' ? `${value}px` : value;
