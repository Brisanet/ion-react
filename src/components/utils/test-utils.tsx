import theme from '@ion/styles/theme';
import { render, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

export const renderWithTheme = (children: React.ReactNode): RenderResult =>
  render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
