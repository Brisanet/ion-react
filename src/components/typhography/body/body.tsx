import React from 'react';
import { IonBodyInterface } from '../typhography.types';
import * as S from './styles';

export const IonBody: React.FC<IonBodyInterface> = ({
  label,
  colorScheme = 'primary',
  size = 'normal',
  style = 'italic',
  weight = 'normal',
}: IonBodyInterface) => {
  const props = { color: colorScheme, size, style, weight };
  return (
    <S.Body {...props} data-testid="ion-body">
      {label}
    </S.Body>
  );
};
