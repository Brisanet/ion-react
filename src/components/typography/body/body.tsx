import { IonBodyInterface } from '../typography.types';
import * as S from './styles';

export const IonBody = ({
  label,
  colorScheme = 'primary',
  size = 'normal',
  style = 'normal',
  weight = 'normal',
}: IonBodyInterface) => {
  const props = {
    $color: colorScheme,
    $size: size,
    $style: style,
    $weight: weight,
  };
  return (
    <S.Body {...props} data-testid='ion-body'>
      {label}
    </S.Body>
  );
};
