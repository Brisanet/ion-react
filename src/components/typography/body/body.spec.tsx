import { renderWithTheme } from '@ion/components/utils/test-utils';
import { screen } from '@testing-library/react';
import { IonBodyInterface } from '../typography.types';
import { IonBody } from './body';

const defaultProps: IonBodyInterface = { label: 'Ion React' };
const sut = (props: IonBodyInterface = defaultProps) => {
  return renderWithTheme(<IonBody {...props} />);
};

const bodyStyles: Array<IonBodyInterface['style']> = ['italic', 'normal'];
const bodySize: Array<IonBodyInterface['size']> = ['small', 'normal'];
const bodyWeight: Array<IonBodyInterface['weight']> = [
  'normal',
  'medium',
  'bold',
];
const bodyColor: Array<IonBodyInterface['colorScheme']> = [
  'primary',
  'secondary',
  'dark-primary',
  'dark-secondary',
];

describe('<IonBody />', () => {
  it('should render the body with the correct label', () => {
    sut();
    expect(screen.getByTestId('ion-body')).toHaveTextContent(
      defaultProps.label
    );
  });
  it.each(bodySize)(
    'should render the body with the correct size %s',
    (size) => {
      const { container } = sut({ ...defaultProps, size });
      expect(container).toMatchSnapshot();
    }
  );

  it.each(bodyWeight)(
    'should render the body with the correct weight %s',
    (weight) => {
      const { container } = sut({ ...defaultProps, weight });
      expect(container).toMatchSnapshot();
    }
  );

  it.each(bodyColor)(
    'should render the body with the correct color %s',
    (colorScheme) => {
      const { container } = sut({ ...defaultProps, colorScheme });
      expect(container).toMatchSnapshot();
    }
  );

  it.each(bodyStyles)(
    'should render the body with the correct styles %s',
    (style) => {
      const { container } = sut({ ...defaultProps, style });
      expect(container).toMatchSnapshot();
    }
  );
});
