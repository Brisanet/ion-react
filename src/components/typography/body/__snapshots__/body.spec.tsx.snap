// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IonBody /> should render the body with the correct color dark-primary 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #e4e6eb;
  font-weight: 400;
  font-style: normal;
}

<div>
  <p
    class="c0"
    data-testid="ion-body"
  >
    Ion React
  </p>
</div>
`;

exports[`<IonBody /> should render the body with the correct color dark-secondary 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #ced2db;
  font-weight: 400;
  font-style: normal;
}

<div>
  <p
    class="c0"
    data-testid="ion-body"
  >
    Ion React
  </p>
</div>
`;

exports[`<IonBody /> should render the body with the correct color primary 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #505566;
  font-weight: 400;
  font-style: normal;
}

<div>
  <p
    class="c0"
    data-testid="ion-body"
  >
    Ion React
  </p>
</div>
`;

exports[`<IonBody /> should render the body with the correct color secondary 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #8d93a5;
  font-weight: 400;
  font-style: normal;
}

<div>
  <p
    class="c0"
    data-testid="ion-body"
  >
    Ion React
  </p>
</div>
`;

exports[`<IonBody /> should render the body with the correct size normal 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #505566;
  font-weight: 400;
  font-style: normal;
}

<div>
  <p
    class="c0"
    data-testid="ion-body"
  >
    Ion React
  </p>
</div>
`;

exports[`<IonBody /> should render the body with the correct size small 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #505566;
  font-weight: 400;
  font-style: normal;
}

<div>
  <p
    class="c0"
    data-testid="ion-body"
  >
    Ion React
  </p>
</div>
`;

exports[`<IonBody /> should render the body with the correct styles italic 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #505566;
  font-weight: 400;
  font-style: italic;
}

<div>
  <p
    class="c0"
    data-testid="ion-body"
  >
    Ion React
  </p>
</div>
`;

exports[`<IonBody /> should render the body with the correct styles normal 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #505566;
  font-weight: 400;
  font-style: normal;
}

<div>
  <p
    class="c0"
    data-testid="ion-body"
  >
    Ion React
  </p>
</div>
`;

exports[`<IonBody /> should render the body with the correct weight bold 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #505566;
  font-weight: 700;
  font-style: normal;
}

<div>
  <p
    class="c0"
    data-testid="ion-body"
  >
    Ion React
  </p>
</div>
`;

exports[`<IonBody /> should render the body with the correct weight medium 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #505566;
  font-weight: 600;
  font-style: normal;
}

<div>
  <p
    class="c0"
    data-testid="ion-body"
  >
    Ion React
  </p>
</div>
`;

exports[`<IonBody /> should render the body with the correct weight normal 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #505566;
  font-weight: 400;
  font-style: normal;
}

<div>
  <p
    class="c0"
    data-testid="ion-body"
  >
    Ion React
  </p>
</div>
`;
